### PR TITLE
Total average rate SLO in the `ledger-api-bench-tool` [DPP-826]

### PR DESCRIPTION
**Pull request by kamil-da**
### Changes in `ledger-api-bench-tool`
- A new stream SLO: maximum item rate (configurable with cli option `max-item-rate` or YAML `max_item_rate`)
- Added a possibility to define both maximum and minimum item rate SLOs in the `CountRateMetric` for the same stream
- Removed returning exit code `1` when SLO violated. After this change only internal errors lead to a non-zero exit-code. We do not use this feature in scripting anymore.
- Refactor: moved SLO implementations to corresponding metric companion objects

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/LedgerApiBenchTool.scala [hunk 1]
```diff
@@ -37,8 +37,6 @@ object LedgerApiBenchTool {
               logger.info(s"Benchmark finished successfully.")
             case Left(error) =>
               logger.info(s"Benchmark failed: $error")
-              // Exit with error status code to facilitate scripting
-              sys.exit(1)
```
Comment (rautenrieth-da) [inline ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/LedgerApiBenchTool.scala:41]: How will the SLOs be used then? Will the caller have to parse the log output to see if the benchmark failed?

File: ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala [hunk 2]
```diff
@@ -32,7 +32,8 @@ class WorkflowConfigParserSpec extends AnyWordSpec with Matchers {
           |         - Foo1
           |         - Foo3
           |    objectives:
-          |      min_item_rate: 123""".stripMargin
```
Comment (rautenrieth-da) [inline ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala:35]: All tests in this file have `min_item_rate` and `max_item_rate` defined. Is the idea that those members are mandatory? If yes, remove the wrapping `Option`. If no, add tests with those members missing.

Related, do you want to support a mode where there are no objectives? I have used the benchtool before to measure the ACS performance where I did not have any objectives in mind. If found it a bit awkward that I had to add the `objectives` struct with some random values in it.

**Review by rautenrieth-da — APPROVED** [review]

File: ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala (hunk 2 above)
Comment (kamil-da) [reply ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala:35]: > All tests in this file have min_item_rate and max_item_rate defined. Is the idea that those members are mandatory? If yes, remove the wrapping Option. If no, add tests with those members missing.

All objectives are meant to be optional. I'll add appropriate tests for cases without objectives - thank you for spotting this 👍 

> Related, do you want to support a mode where there are no objectives? I have used the benchtool before to measure the ACS performance where I did not have any objectives in mind. If found it a bit awkward that I had to add the objectives struct with some random values in it.

I'll make them truly optional in configuration as well.

Comment (kamil-da) [reply ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/config/WorkflowConfigParserSpec.scala:35]: Optional objectives: 2523e32f9509060ab1e7d791dd53c2da19befffb
Unit tests: c24d6410b72162233776fd3a5ee4ae63d099a4a5

**Pull request merged**